docs/writing_review_screen.md

```diff
@@ -0,0 +1,240 @@
+code: |
+  review_form
+```
+
+and I'd recommend changing the YAML to how you normally write it, to be more concise [^1] The remainder of the examples will use the more concise YAML notation.
+
+There are a few sections of the generated review screen YAML:
+
+1. There are the events. These blocks look like this:
+    ```yml
+    event: review_intro
+    code: |
+      review_form
+    ---
+    event: review_prior_case
+    code: |-
+      review_form
+    ---
+    ...
+    ```
+
+    These blocks take each section from your interview and make it so that if a user clicks on them, they are brought to the review page. In our experience, this
+    works best, so that all of the information that is asked for in the interview can be edited in the same place, and the user doesn't have to remember
+    which section they entered it in.
+
+2. The main review block. The block looks like this:
+
+    ```yml
+    id: review screen
+    event: review_form
+    question: |
+      Review your answers
+    review:
+      - Edit: users.revisit
+        button: |
+          **Users**
+
+          % for item in users:
+          - ${ item }
+          % endfor
+      ...
+      - Edit: has_existing_case
+        button: |
+          **Do you have an existing Michigan family court case?**
+
+          Do you have an existing Michigan family court case?: ${ word(yesno(has_existing_case)) }
+    ```
+
+    Most notably is that there is a `review` attribute instead of a `fields`. Each element under the `review` attribute
+    is a different section, generated from screens in the interview, with the subquestion and values of the fields shown.
+
+    Each element in the review can have the following dictionary attributes:
+
+    * `Edit` adds an "Edit" button next to the information on the review screen. It's value is the name of the variable that will be edited when the button is clicked.
+    * `label` lets you change the text on the "Edit" button
+    * `fields` lets you add multiple fields that are re-asked to the user for a section on the review screen.
+      * `recompute` lets you also re-calculate other variables, not necessarily re-asking them, but letting them be filled by `code` blocks. These won't be generated, but will come in handy later.
+    * `button` is the actual content of that section of the review screen, similar to a `subquestion` attribute.
+
+3. The revisit screens. These blocks look like this:
+
+    ```yml
+    id: revisit users
+    continue button field: users.revisit
+    question: |
+      Edit your answers about Users
+    subquestion: |
+      ${ users.table }
+
+      ${ users.add_action() }
+    ---
+    ...
+    ```
+
+    Each block has the table and lets people edit items in the lists.
+
+4. The generated tables for the lists.
+
+    ```yml
+    table: children.table
+    rows: children
+    columns:
+      - Birthdate: |
+          row_item.birthdate if hasattr(row_item, 'birthdate') else ''
+    edit:
+      - birthdate
+    ```
+
```

Comment:
> I think this should start by explaining something about the purpose--that these two sections are about how you can edit lists using the review screen approach, and why it might be useful to do so

I agree that more context is needed, but since these are already a part of the "What the generator makes for you" section, I'll stick with describing what they'll see in their generated review screen before going into the what or why.

> which are just lists of people, and maybe the ones that they can delete. Probably we should delete those in the review screen generator though.

Reminds me that there's some upstream changes I need to make to Assembly Line, we don't have any `revisit` screens for the people lists that we define, which we should definitely have.

Reply:
Made those upstream changes (PR incoming), and I took a shot at adding more context to this section.